File: csharp/acronym/Acronym.cs
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        char[] separators = {' ', '-'};
        var acronym = string.Concat(phrase
                .Split(separators)
                .Where(word => word.Any(char.IsLetter))
                .Select(word => word.First(char.IsLetter)))
            .ToUpper();
        return acronym;
    }
}

File: elixir/acronym/lib/acronym.ex
defmodule Acronym do
  @doc """
  Generate an acronym from a string.
  "This is a string" => "TIAS"
  """
  @spec abbreviate(String.t()) :: String.t()
  def abbreviate(string) do
    string
    |> String.split(~r/\s|-/, trim: true)
    |> Enum.map(fn word ->
      word
      |> String.replace(~r/[^[:alpha:]]/, "")
      |> String.first()
    end)
    |> Enum.join()
    |> String.upcase()
  end
end
